plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,216 @@
+    private final List<DeltaLakeColumnHandle> partitionColumns;
+    private final List<RowType.Field> partitionFields;
+
+    public DeltaLakePartitionsTable(
+            ConnectorSession session,
+            SchemaTableName tableName,
+            String tableLocation,
+            TransactionLogAccess transactionLogAccess,
+            TypeManager typeManager)
+    {
+        this.tableName = requireNonNull(tableName, "tableName is null");
+        this.tableLocation = requireNonNull(tableLocation, "tableLocation is null");
+        this.transactionLogAccess = requireNonNull(transactionLogAccess, "transactionLogAccess is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+
+        this.partitionColumns = getPartitionColumns(session, new SchemaTableName(tableName.getSchemaName(), DeltaLakeTableName.tableNameFrom(tableName.getTableName())), this.tableLocation);
+
+        this.partitionFields = this.partitionColumns.stream()
+                .map(column -> RowType.field(column.getBaseColumnName(), column.getType()))
+                .collect(toImmutableList());
+
+        ImmutableList.Builder<ColumnMetadata> columnMetadataBuilder = ImmutableList.builder();
+
+        if (!this.partitionFields.isEmpty()) {
+            columnMetadataBuilder.add(new ColumnMetadata("partition", RowType.from(this.partitionFields)));
+        }
+
+        columnMetadataBuilder.add(new ColumnMetadata("file_count", BIGINT));
+        columnMetadataBuilder.add(new ColumnMetadata("total_size", BIGINT));
+
+        this.tableMetadata = new ConnectorTableMetadata(tableName, columnMetadataBuilder.build());
+    }
+
+    @Override
+    public Distribution getDistribution()
+    {
+        return Distribution.SINGLE_COORDINATOR;
+    }
+
+    @Override
+    public ConnectorTableMetadata getTableMetadata()
+    {
+        return tableMetadata;
+    }
+
+    @Override
+    public ConnectorPageSource pageSource(ConnectorTransactionHandle transactionHandle, ConnectorSession session,
+                                          TupleDomain<Integer> constraint)
+    {
+        TableSnapshot tableSnapshot;
+
+        try {
+            // Verify the transaction log is readable
+            SchemaTableName baseTableName = new SchemaTableName(tableName.getSchemaName(),
+                    DeltaLakeTableName.tableNameFrom(tableName.getTableName()));
+            tableSnapshot = transactionLogAccess.loadSnapshot(session, baseTableName, tableLocation);
+        }
+        catch (IOException e) {
+            throw new TrinoException(DELTA_LAKE_INVALID_SCHEMA,
+                    "Unable to load table metadata from location: " + tableLocation, e);
+        }
+
+        if (partitionColumns.isEmpty()) {
+            return new EmptyPageSource();
+        }
+        return new FixedPageSource(buildPages(session, tableSnapshot));
+    }
+
+    private List<Page> buildPages(ConnectorSession session, TableSnapshot tableSnapshot)
```

Comment:
The implementation for column mapping mode seems broken. Please fix and add a test:
```sql
create table test (id int, part int) with (column_mapping_mode = 'id', partitioned_by = array['part']);
insert into test values (1, 2);
select * from "test$partitions";
Cannot invoke "java.util.Optional.map(java.util.function.Function)" because "valueString" is null
```

Reply:
TestDeltaLakeSystemTables#testColumnMappingModePartitionsTable